authentication/account/service/user.go

```diff
@@ -0,0 +1,127 @@
+package service
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/fabric8-services/fabric8-auth/application/service"
+	"github.com/fabric8-services/fabric8-auth/authorization/token/manager"
+
+	"github.com/fabric8-services/fabric8-auth/application/service/base"
+	servicecontext "github.com/fabric8-services/fabric8-auth/application/service/context"
+	"github.com/fabric8-services/fabric8-auth/authentication/account/repository"
+	"github.com/fabric8-services/fabric8-auth/errors"
+	"github.com/fabric8-services/fabric8-auth/log"
+	"github.com/satori/go.uuid"
+)
+
+// NewUserService creates a new service to manage users
+func NewUserService(ctx servicecontext.ServiceContext) service.UserService {
+	return &userServiceImpl{
+		BaseService: base.NewBaseService(ctx),
+	}
+}
+
+// userServiceImpl implements the UserService to manage users
+type userServiceImpl struct {
+	base.BaseService
+}
+
+// UserInfo gets user information given a context containing access_token
+func (s *userServiceImpl) UserInfo(ctx context.Context, identityID uuid.UUID) (*repository.User, *repository.Identity, error) {
+	var identity *repository.Identity
+	err := s.ExecuteInTransaction(func() error {
+		var err error
+		identity, err = s.Repositories().Identities().LoadWithUser(ctx, identityID)
+		if err != nil || identity == nil {
```

Comment:
if no `identity` is found, the repository method already returns an error (https://github.com/fabric8-services/fabric8-auth/blob/master/account/repository/identity.go#L178), so there's no need to check for `identity == nil` I believe

Reply:
This code was refactored as-is, although I have no strong objection to updating it in this PR.